jmix-multitenancy/multitenancy/src/main/java/io/jmix/multitenancy/core/impl/TenantEntityOperationImpl.java

```diff
@@ -79,9 +84,11 @@ private Field findTenantField(Class<?> entityClass) {
     public void setTenant(Object entity, String tenantId) {
         MetaProperty property = findTenantProperty(entity.getClass());
         if (property != null) {
-            EntityValues.setValue(entity, property.getName(), tenantId);
+            Object reloadedEntity = dataManager.load(Id.of(entity))
+                    .fetchPlanProperties(property.getName())
+                    .one();
```

Comment:
Anyway, this code will probably fail when you try to create a new tenant instance in a regular way (as it is described in the documentation)